src/test/java/uk/gov/hmcts/divorce/systemupdate/service/task/RegenerateConditionalOrderPronouncedCoversheetTest.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.hmcts.divorce.systemupdate.service.task;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.verify;
+import static uk.gov.hmcts.ccd.sdk.type.YesOrNo.YES;
+import static uk.gov.hmcts.divorce.divorcecase.model.ApplicationType.JOINT_APPLICATION;
+import static uk.gov.hmcts.divorce.divorcecase.model.ApplicationType.SOLE_APPLICATION;
+import static uk.gov.hmcts.divorce.divorcecase.model.ContactDetailsType.PRIVATE;
+import static uk.gov.hmcts.divorce.document.model.DocumentType.CONDITIONAL_ORDER_GRANTED;
+import static uk.gov.hmcts.divorce.document.model.DocumentType.CONDITIONAL_ORDER_GRANTED_COVERSHEET_APP_1;
+import static uk.gov.hmcts.divorce.document.model.DocumentType.CONDITIONAL_ORDER_GRANTED_COVERSHEET_APP_2;
+import static uk.gov.hmcts.divorce.testutil.TestDataHelper.buildCaseDataCOPronounced;
+
+@ExtendWith(MockitoExtension.class)
+public class RegenerateConditionalOrderPronouncedCoversheetTest {
+
+    @Mock
+    private GenerateConditionalOrderPronouncedCoversheet generateCoversheet;
+
+    @InjectMocks
+    private RegenerateConditionalOrderPronouncedCoversheet underTest;
+
+    @Test
+    public void shouldGenerateCoverLettersSoleApplication() {
```

Comment:
I think there needs to be a few unhappy path tests to cover some of the if conditions

Reply:
added more tests